components/shared/script_layout/wrapper_traits.rs

```diff
@@ -285,9 +285,12 @@ pub trait ThreadSafeLayoutNode<'dom>: Clone + Copy + Debug + NodeInfo + PartialE
 
     fn node_text_content(self) -> Cow<'dom, str>;
 
-    /// If the insertion point is within this node, returns it. Otherwise, returns `None`.
+    /// If there's selection within this node, returns it. Otherwise, returns `None`.
     fn selection(&self) -> Option<Range<ByteIndex>>;
```

Comment:
This should probably return an enum specifying whether the selection totally overlaps of the node or whether it is only a range within the node. Then you can get rid of `insertion_point`.

Reply:
https://github.com/servo/servo/pull/35830#discussion_r1988980059
nvm I saw the other comment which makes sense.